### ISSUE TITLE
Add optional vendor creation step to projects

### ISSUE THREAD

**Issue by frjo** [HyphaApp/hypha#2379]
This story is part of epic #2352.

> As a contractor I need to provide my banking, tax information to the funder so they can setup me up in their accounting system.

**What the user needs to do**
In Hypha the contractor needs to submit this information:

1. name of organisation or person mentioned on the contract with OTF
2. name of person with authority to sign the contract
3. upload completed w8 or w9 form (depending on their US tax status)
4. bank account holder name
5. bank account number
6. bank account routing number
7. bank account currency
It needs to provide the name and abbreviation. Screenshot from the designs:
![image](https://user-images.githubusercontent.com/1899506/118988419-15a13580-b979-11eb-974c-7f5d5ef3886d.png)

8. address of the bank where the account is located
[A country picker, preferably working like this](https://baymard.com/labs/country-selector)
![Kapture 2021-05-20 at 15 43 59](https://user-images.githubusercontent.com/1899506/118989477-f951c880-b979-11eb-9266-8e85d07d23c8.gif)

9. provide extra information to receive payments if needed - [full information is in the designs](https://www.figma.com/file/6HzFAw7k9zXwuujvyf5uVO/OTF-System?node-id=192%3A2972)

**Designs**
These designs show how the interface should work:
https://www.figma.com/file/6HzFAw7k9zXwuujvyf5uVO/OTF-System?node-id=192%3A2972

**List**
- [ ] Add a setting in "Project settings" (already exist) to turn vendor creation on/off. If on it will be a required step for the applicant/vendor.
- [ ] When vendor creation is on it is a required step before contract and invoicing can happen.
- [ ] On MESSAGES.CREATED_PROJECT send e-mail to applicant informing them that vendor setup is needed.
- [ ] Create a form for adding vendor information. For fields and UX see below.
- [ ] Create a vendor in third party service if exist
- [ ] Update vendor status from third party service.
- [ ] Allow applicant and staff to edit the vendor information.
- [ ] Notify lead when vendor information has been added.

The vendor information like bank account might change over time of the project. We need to keep track of this somehow. I see two options:

1. Have revisions on the vendor model och link revisions to invoices.
2. Copy current vendor information to the invoice and keep the history there.

In a separate app that is OTF only we will be making API calls to create the vendor in a third party system.

**Comment by ei8fdb** [comment]
@frjo @Parbhat 

Here's the designs for the setup new contractor - 
https://www.figma.com/file/6HzFAw7k9zXwuujvyf5uVO/OTF-System?node-id=192%3A2972

I'll add the step for the contractor to access this page when I get a minute.

Are there any challenges in these designs? 

Any questions, thoughts, please let me know.

**Comment by sks444** [comment]
@frjo @ei8fdb 

Thanks for the details and the design!

Have designed the following database model for vendor looking at the design. Let me know your suggestions and if something is missing.

Vendor: 
    - name: Char Field
    - contractor name: Char Field
    - type: Choice Field (Organization, Personal)
    - required to pay taxes: Boolean Field
    - w8 form: File Field
    - w9 form: File Field
    - bank info: One To One relation with BankInformation model
    - other info: Text Field

BankInformation:
    - account holder name: Char Field
    - account routing number
    - account number
    - account currency: is this choice field?
    - needs extra info: Boolean Field
    - branch address: we can use the address field that we have.
    - intermediary account info: one to one with self
    - nid type(National Identity Document)
    - nid number

- There is a custom [Address field](https://github.com/HyphaApp/hypha/blob/e181ebadfb744aab34617bb766e746368d6f2de0/addressfield/fields.py#L33) which we use throughout Hypha. It provides the following fields, which has the country picker similar to what we need. Looks like we can directly use this instead of separate address fields?

<img width="365" alt="Screenshot 2021-05-27 at 4 52 59 PM" src="https://user-images.githubusercontent.com/22263040/119826576-cb3e2c80-bf15-11eb-9ea7-3df0d57ae769.png">



- Is the currency, choice form field or we want this to work similar to country?

> 1. Have revisions on the vendor model och link revisions to invoices.
> 2. Copy current vendor information to the invoice and keep the history there.

Both looks promising. Will check which one is more feasible to track with invoices.

**Comment by sks444** [comment]
Few questions related to design:

In the design, vendor creation process is happening on the project detail page/details tab.

- I am wondering where would we show the project details in that time period?
- Where are we showing details of vendor information once that is done?
- Where will the editing of vendor happen?

I am thinking what if we have this feature similar to `Add Review/Add Determination` on submission.

- a button for vendor creation in `Actions to take` sidebar whenever it is required. (Other feature like contracting/invoicing can be disabled/invisible until that is done.).
- On click of the button open a new page with form, where user continues with multi-step vendor creation process.
- A side bar for `Vendor Information`.. and based on the state of the process user can Add/Edit/View vendor information.

**Comment by frjo** [comment]
The currency field can be a simple dropdown with all the currencies. Would be good to use a python/django package for this so we do not have to maintain a list of currencies.

**Comment by frjo** [comment]
In the current version when staff approves a project they add what a lot of what we now consider vendor information, e.g. address. This end up in "application_projects_project".

Reuse this model or move it all to a new separate vendor model? We do not want to have duplication in any case.

**Comment by ei8fdb** [comment]
Hello @sks444 I think its easier if I put together a design to answer your questions above. I'll post it here shortly. If you still have questions, let me know. OK?

**Comment by sks444** [comment]
Hey @ei8fdb, sure, thanks!

**Comment by sks444** [comment]
> Reuse this model or move it all to a new separate vendor model? We do not want to have duplication in any case.

Would be good to move vendor specific fields(user/[contact fields](https://github.com/HyphaApp/hypha/blob/84e05a1b1d6b20db53e2e619869fbde6feffc11d/hypha/apply/projects/models.py#L363)) to vendor model. Btw, do we need to do data migration for old projects, I suppose we are not using the projects features currently, so it won't be needed?

**Comment by frjo** [comment]
No need for migrations. Feel free to add/delete/move stuff around.

**Comment by frjo** [comment]
@sks444 No clear decision  on where OTF really want to store sensitive information like bank details (good idea making that a separate model).

Vendor information will for OTFs case be stored in IntAcct. If a third party system exist for vendor can we have Hypha always use that information? Even scrubbing bank and other info from Hypha DB after the vendor has been created in the Third party system?

Some Hypha users will not have any third party service so they would need to store the info in Hypha however.

Sounds complicated just describing it.

**Comment by sks444** [comment]
Good points @frjo. Thanks for bringing it up.

Yes, the general process with the payment systems is to have the forms/interface for required inputs and directly store data to third party system on submit..storing just an id of that vendor in our system for further communication.

But given that Hypha itself acts like a package, being open to support any third party payment systems. It sounds okay to have db-models, and give hypha users the flexibility of storing that data on the system of their choice/maybe even hypha.

Once all the needed data is stored to third party systems, it should be deleted from ours, and the further process like, viewing/editing should directly be done with the api.

We can have a field on vendor(maybe status, with choices, pending/active etc) which tells us all the required data has been full-filled and it could be deleted from our system. And then run a cronjob which checks for that field and accordingly deletes the sensitivity information. This field can also help in editing feature, once it is true, maybe we should not allow editing the data we deleted, for the MVP. _As this editing would require direct communication with API._ **[Could be discussed later.]**

Given that we have a cron job for the transfer of data mentioned in https://github.com/OpenTechFund/apply-app/issues/1, which also updates the `status` field on a vendor accordingly.

**Comment by frjo** [comment]
@sks444 We do not need separate uploads fields for w8 and w9. Non US residens fill in w8 and US residens w9.

A more generic "project due-diligence documents" field that can store one or more documents per vendor would be best.

The whole w8/w9 thing is only interesting for US based organisations. So let's complicate vendor even more :-).

Can we have a setting that toggles between the "US taxes" form step and a more generic "upload needed documents" section.

Lets call the section "Due-diligence documents" and store the uploads files in the same db table. Add also a setting allowing a organisation to add text here to specify what they need from the contractor.

**Comment by ei8fdb** [comment]
> Hey @ei8fdb, sure, thanks!

@sks444 Here's a short video on how the new contractor navigates to the new contractor setup form, and what happens when they have completed it.

https://user-images.githubusercontent.com/1899506/120212222-a1dd1380-c229-11eb-9667-740a5b7270db.mp4


The associated wireframes are available on [balsamiq cloud here](https://balsamiq.cloud/so3r3ka/p9tuu7g/rE446?f=N4IgUiBcAMA0IDkoEZ4GEXwDKZAISjhAFlD4AtXAaSgG1QAnKEAMQE4ARADhAF9ZGzdgEEALHwEgmkEGwCiAdmgTBMvAuHIVU5nI4cM%2FVSDmi5PIzpnsAbACZt0kAFYOch5acLuhyU5t40ASezDZsaADMjszCiizRMmEcwX7MXHLIWiEyEchoNgkueBxR2SbOEeJlEc5cvsZybMIKhSx2eHKFdhEsFqkywiy5hWz2KcY2yc6FaM6RhcjO0MKFCs52Hv34aFylW6LCzllbXNBsmw0cQ11sCuNWIKJsT4VcyND1D3nsC%2BwchXJ1PEylxnKILg89DYCmVkHZRPcnHMIv8ymwIlw%2BsY7Bx0as8EtCjZARCnHIesCtsJYitYWhoHtjMszq1kODCs5OIjmAdkoU8DZvESBbSqXY5oVRAo2McJm42K1hDYqltoOYFWUrtBUVtkG9ZQ8IpNpmiItAVQ07GyFmadcY8IMDU47GgrV1FAaAEY84TnEDwLwE5SWb0yA5%2BgO6CnaUOPX0OSMyLV2kCxuRoPBaRMgZN8AC6vCAA%3D%3D) marked as "Scenario 1.a A new contractor needs to complete the "setup new contractor" form for the first time" to "Scenario 1.e New contractor setup form completed".

Let me knw if you have any other questions.

**Comment by sks444** [comment]
@frjo, some of the fields mentioned in the design have hard coded name of the organisation using Hypha. E.g. `OTF` in the first field.."1. What is the name of the person/organisation on the contract with OTF?". 

So I guess we would have to have a form setting where we set custom label and help text for all the fields. Similar to what we had for [determination](https://github.com/HyphaApp/hypha/blob/a64e0eff7fa9a3c424af825d90de4d07bb632a8b/hypha/apply/determinations/models.py#L232)...on that note _I just wanted to confirm that we are ruling out the possibility of creating setup contractor forms with Wagtail(Although this doesn't look preferable, given that this is a multistep form/storing sensitive information and would bring more complexity to the system)._

**Comment by frjo** [comment]
@sks444 There are variables available for organisation name, ORG_LONG_NAME and ORG_SHORT_NAME.

Also see https://github.com/HyphaApp/hypha/issues/2408 regarding changing/translating strings.

Allowing Hypha users to create these forms in Wagtail just as they create application forms etc. would be a nice feature. But with API integration and the issues you mention above I think it would take a lot of work to implement.

**Comment by sks444** [comment]
Hello Team,

I have been working on this for some days now. So thought it would good to update on the progress a bit. 

I am using [django-formtool's wizard form](https://django-formtools.readthedocs.io/en/latest/wizard.html) for creating multi-step forms. Using [SessionWizardView](https://django-formtools.readthedocs.io/en/latest/wizard.html#formtools.wizard.views.SessionWizardView), so the data will be stored in session until the form is submitted in final step.

Have managed to create forms for each step mentioned in the design. Any of the field's label/helptext can be edited from [VendorFormSettings](https://github.com/HyphaApp/hypha/blob/48ea8861e76a6a39b9503d09eb31917d471d8dff/hypha/apply/projects/models/vendor.py#L91). I'm also able to save data on the form submission in final step.

However I'm facing a few challenges that needs to be resolved:

- Not able to load django-file-form drag and drop ui for the form step where we need file uploads(due diligence documents).
- AddressField that we use in hypha breaks when there are two of them on the same page(last step in the form)

Along with that, looks like the last form step is conditional..so if we select `No`..in the step `Do we need to provide us with extra information?`, that should be the final step and next form step shouldn't be shown.

I am targeting these to be fixed by the end of this week. Once the creation of vendor steup is done we can move to editing of it.

**Comment by ei8fdb** [comment]
> Hello Team,

Hello @sks444 👋 
I'll answer the design related questions, and I think it's better for me to leave the code questions for frjo. :)
 
> so the data will be stored in session until the form is submitted in final step.
> Have managed to create forms for each step mentioned in the design. Any of the field's label/helptext can be edited 
💯 👍 

> the last form step is conditional..so if we select `No`..in the step `Do we need to provide us with extra information?`, that should be the final step and next form step shouldn't be shown.

Exactly, you're right if the user selects "No", that should be the final step and next form step shouldn't be shown.

> I am targeting these to be fixed by the end of this week. Once the creation of vendor steup is done we can move to editing of it.

👍 I'm looking forward to seeing it on my screen! Is there anything else you need from me?

**Comment by sks444** [comment]
Hey @ei8fdb, thanks! I'll let you know if there is anything. :)

**Comment by frjo** [comment]
@sks444 Would using a css/js solution to make the form look lika a multi step form when it really just is a single HTML page simplify the implementation?

**Comment by frjo** [comment]
@ei8fdb This form is filled in once per project and they are highly motivated to fill it in or they get no money. If having it multistep makes the implementation hard could we simplify it to a single standard form?

**Comment by ei8fdb** [comment]
> @ei8fdb This form is filled in once per project and they are highly motivated to fill it in or they get no money. If having it multistep makes the implementation hard could we simplify it to a single standard form?

Creating multi-page forms is an important design pattern for Hypha. Can we have a chat about it in the morning? Can we investigate what might be difficult to implement?

There is a lot going on in this form and good design practice is to break up the form to focus on each part section by section. Let's see what's possible.

**Comment by sks444** [comment]
@frjo, @ei8fdb, most things here are done. Let me try and fix the remaining issues. Once we have something ready for testing on Monday. Based on that we can tweak design/implementations.

**Comment by ei8fdb** [comment]
> @frjo, @ei8fdb, most things here are done. Let me try and fix the remaining issues. Once we have something ready for testing on Monday. Based on that we can tweak design/implementations.

@sks444 if having a video call makes it easier to discuss just say and we can do that. Whatever is easier.

**Comment by sks444** [comment]
**Update:**

>Not able to load django-file-form drag and drop ui for the form step where we need file uploads(due diligence documents).

This has been fixed.

>looks like the last form step is conditional..so if we select No..in the step Do we need to provide us with extra information?, that should be the final step and next form step shouldn't be shown.

Fixed.

>AddressField that we use in hypha breaks when there are two of them on the same page(last step in the form)

This remains. It works properly with one address field. But when there are two of them it doesn't work properly.

Along with that I just have to do some cleanup and this will be ready for testing. Will update here.

**Comment by sks444** [comment]
**Update:**

I have a few more things to do however this is ready for testing now.

https://test-apply.hypha.app/apply/projects/28/

Quick notes:

- Submit an application, and approve, so that a project is created(Or test on a [project](https://test-apply.hypha.app/apply/projects/28/) that I created.). (we initialise few of the vendor info on project creation)
- Visit project detail page, in the supporting documents section we can see, `Contractor Setup Form`, click on `Update info`.
- Form field's label/help_text can be updated from the [Vendor Form Setting](https://test-apply.hypha.app/admin/settings/application_projects/vendorformsettings/2/) in wagtail admin.
- We are using only a single field for W8/W9 forms("Due diligence documents")..accordingly we can add help text for this field for user to know what to upload.
- Note that the Intermediary Bank Branch Address field on the last step of the form is missing for now. As we don't have support for two AddressField on the same form page. Will add it once fixed. https://github.com/HyphaApp/hypha/issues/2426


Let me know your feedback and if something is breaking. Thanks!

**Comment by fourthletter** [comment]
This feature looked like it took an enormous amount of work, thank you for shepherding this feature into reality.  I will test these features further with Finance later this week. 

The current pick list on test offers German Mark and Italian Lira as options. Is there something going on with Euro that I'm not aware of? Euro is also an option. 

List

- [x] Add a setting in "Project settings" (already exist) to turn vendor creation on/off. If on it will be a required step for the applicant/vendor.
- [x] When vendor creation is on it is a required step before contract and invoicing can happen.
- [ ] On MESSAGES.CREATED_PROJECT send e-mail to applicant informing them that vendor setup is needed.
- [x] Create a form for adding vendor information. For fields and UX see below.
- [ ] Create a vendor in third party service if exist
- [ ] Update vendor status from third party service.
- [x] Allow applicant and staff to edit the vendor information.
- [ ] Notify lead when vendor information has been added.

_The vendor information like bank account might change over time of the project. We need to keep track of this somehow. I see two options:_

- [ ] Have revisions on the vendor model och link revisions to invoices.
- [ ] Copy current vendor information to the invoice and keep the history there.
- [ ] In a separate app that is OTF only we will be making API calls to create the vendor in a third party system.

Vendor Detail Page
- [x] click on view to visit vendor detail page
- [x] detail vendor page has 'edit' option
- [x] Indicate when the form has been completed. For example, in this [submission on test](https://test-apply.hypha.app/apply/projects/29/), the form was completed but there is NO indication that the information has been saved or is available. 

<img width="486" alt="Screen Shot 2021-06-30 at 2 10 50 PM" src="https://user-images.githubusercontent.com/20019656/123958324-34014480-d9ad-11eb-90dd-4c0108ab8ec1.png">

**Comment by sks444** [comment]
Thanks @fourthletter, for testing it out.

--

@frjo, I believe we want to skip the step, "Is the organization required to pay US taxes?" based on a form setting https://github.com/HyphaApp/hypha/issues/2379#issuecomment-851495865? As not all organisation using Hypha will have that question. Let me know if that is needed, I have a change which does that.

Along with that I have the following changes in my list that I will do today:
>Add a setting in "Project settings" (already exist) to turn vendor creation on/off. If on it will be a required step for the applicant/vendor.

- [ ] I think we need to add another field in Project Settings, `required vendor setup`, and based on that display the option to setup vendor on the project detail page?

- [ ] On MESSAGES.CREATED_PROJECT send e-mail to applicant informing them that vendor setup is needed.

> Notify lead when vendor information has been added.

Do we need to do this every time vendor info gets updated OR just when the vendor setup is completed? This also raises the point of: _When do we know that vendor setup is completed?_ As mentioned in https://github.com/HyphaApp/hypha/issues/2409, this could be known when we get an ok response from the third party payment service. If the notify needs to happen on completed, we will do this when building the cron jobs.

--

I am planning to look into third party integration/invoicing once we have merged the first version of vendor setup. Will also add the support for revision in another pr(maybe when looking into the invoicing).

As most of the features are dependent on each other, the implementation plan is to have the first version of each feature ready and improve on them as we proceed further in the development of projects app.

**Comment by frjo** [comment]
@sks444  All organisation we talked to that use Hypha has some forms/documents that must be uploaded by the vendor. For US it is w8/w9 for other countries other forms, but always something.

**Comment by ei8fdb** [comment]
Hi @sks444 @frjo asked me to add these designs for how a partner (previously called vendor or contractor) would get to update/modify the setup form.

### Requirements
The partner has completed the partner set-up but now needs to update it with some new information. They need to:
- be able to find it from their project page
- see what information was completed before
- see when it was last updated
- be able to update the information
- then get back to their project page

![Scenario 8 Introduction](https://user-images.githubusercontent.com/1899506/122886677-750bb000-d340-11eb-93e4-3d243e5c48e5.png)

### When the project partner logs in
The partner starts by logging into Hypha. They see the project home page. On the right hand side of the page they can see the link _partner setup form_.

They click on this and are brought to the next page.

![8 1](https://user-images.githubusercontent.com/1899506/122886656-7046fc00-d340-11eb-9687-0a01eaa58048.png)

### Display the partner setup form information
When the partner clicks on the “Partner setup form” link on the previous page, they see the partner setup information they’ve entered before.

At the top of the page they see when this was last updated. They see all the information that was entered. It is displayed as text.

If they want to update this information they can click on “Edit partner information” button.

![8 2](https://user-images.githubusercontent.com/1899506/122886665-72a95600-d340-11eb-8e1e-7c9cff9a9d68.png)

### Display the partner setup form information
When they click the “Edit partner information” button they are brought to the partner setup form as shown in the [Figma design as shown above](https://www.figma.com/file/6HzFAw7k9zXwuujvyf5uVO/OTF-System?node-id=192%3A2972). 

![8 3](https://user-images.githubusercontent.com/1899506/122886673-73da8300-d340-11eb-9779-4d76689048b2.png)

### After they modify they return to their project page
When the partner has finished updating the information, they are brought back to project home page.
![8 4](https://user-images.githubusercontent.com/1899506/122886675-74731980-d340-11eb-8459-86b059ff8388.png)


You can try out a clickable prototype using[ this link](https://balsamiq.cloud/so3r3ka/p9tuu7g/r76A5?f=N4IgUiBcAMA0IDkoEZ4GEpxAGRfAQpvALJ4gBaRIA0lANqgBOUIAYgJwAiAHCAL6wmLDgEEALP0EhmkEOwCiAdmiShs%2FIpHJV0lvM6cMAtSHlj5vY7tkcAbACYdMkAFZO8x1eeKeRqc9t8aEIvFlt2NABmJxYRJVYY2XDOEP8WbnlkbVDZSOQ0W0TXfE5onNN2EUUi1nt8eSL7SNZLNNl2B1STWxSXIrQXKKLuTjFPNpBIl24%2FE3kXSIlykVY8opdoaHYixRd7cZN8NG4yibERF2yJ7i2D61NOVcb2RS77sXYP4eRoWfv8jhFZAcThFeQaBLlbguMZgzi2QrlZD2MRvZyDSKg8rsSLcVomeycHE7fAbIq2cF3ZzyZqQiYiOIiIFoaCnEzQERbGrIWHlFxcNEsc4pIr4Ww%2BclipnleTQbhXOa4%2FH3cGfIrsF4NcqcexDJHyA1AhZUljsVhiBX3FyrFTLEQafoiSJs%2B4iXVYs6vS3oy5%2FZxoZAO8rQCzbbWsaAekzIeXeliRHp9bGRaBLCYeHlAlNR%2B74FZxpIJv0sXX2AsgBFBiaVETK5xiFyXdbQViIiYuRQjI2jIqcJ1tkwd7hpkz2xQm2TQRTILX02ysYuyexm6XVjID%2B72JTlsxiDfOH7KHb2SNFMSlQWyR6cJMTWwfS9yTl00AAI3jm208APU9tUnfuSfiA35CheOgASA5zsI4IGyK8ZJWBBUEwSqtLgUKIjQcB9yPKeiEYVhsEVkW6GyPIaD4F%2BOERjmEHkZR2EBCR%2BFXiM0REY2%2BSkSAjzINsHG%2Btx8jiuxVqCSxkEMkyRFiF6QngoUMlyRJ%2BCqYprruvwAC6fBAA%3D).

@sks444 if you have any questions, let me know!

**Comment by sks444** [comment]
New changes has been deployed on test.

We can click on view to visit vendor detail page: 
<img width="630" alt="Screenshot 2021-07-02 at 10 00 29 AM" src="https://user-images.githubusercontent.com/22263040/124220983-7006e780-db1c-11eb-89e9-8ce799fa9ed6.png">


Added field to activate/deactivate vendor setup: https://test-apply.hypha.app/admin/settings/application_projects/projectsettings/2/

We are also sending email on project creation to mention that vendor setup is required.

**Comment by fourthletter** [comment]
@sks444 I just tested, looks great! The little blue check does not indicate that the form has been completed. However, it is NOT a major issue, this feature completes me. 💥

**Comment by sks444** [comment]
Thanks @fourthletter!

>The little blue check does not indicate that the form has been completed 

Yes, I have that in my TODO. Actually for now, there is no certain way to know when the form is completed. One way to mark it completed is when we successfully have updated the info at third party payment system. I.e. IntAcct for OTF.

**Or**

We can mark it as completed whenever user update info for the first time..as that means user has filled all the required information.

@frjo, what do you suggest?

**Comment by ei8fdb** [comment]
Thanks @fourthletter and @sks444 for noticing the blue tick. I forgot to mention that. 

Probably silly question - is it not possible to detect when the user presses the `submit button` at the end of the setup form?

A few questions @sks444:

> Yes, I have that in my TODO. Actually for now, there is no certain way to know when the form is completed. One way to mark it completed is when we successfully have updated the info at third party payment system. I.e. IntAcct for OTF.

How would this work for an organisation not using a 3rd party finance system?

> **Or**
> 
> We can mark it as completed whenever user update info for the first time..as that means user has filled all the required information.

And if the user needed to change some information, they could click "Update info"?

**Comment by fourthletter** [comment]
I prefer the **second option**: "We can mark it as completed whenever user update info for the first time..as that means user has filled all the required information."

The reason I prefer the **second option** is because the hypha's project page is where OTF engage and communicate with the vendor/contractor. 

- The **vendor/contractor** needs to know whether they have not completed the form and/or edit a form that has previously been completed/submitted to OTF.
- **Compliance/OTF staff** needs to know this step has been completed before they can initiate the contract process. 

If there is a technical glitch with transferring information between hypha and intacct, the vendor and compliance staff are not required to resolve this issue. This means the contractor set-up feature shouldn't be where a glitch is conveyed.

**Comment by frjo** [comment]
@sks444 I also think the second option is preferable. Maybe something like https://github.com/HyphaApp/hypha/blob/d31faa57a76a08dd2be53ff9b26a29a1e4d3d372/hypha/apply/projects/forms/project.py#L112

**Comment by sks444** [comment]
Agree!

As all the fields are required in the form(except for the last step which is optional). This wouldn't be a problem.

I will make this change.

>is it not possible to detect when the user presses the submit button at the end of the setup form?

Yes, @ei8fdb, it is possible. That is what I will do now. Only reason I was avoiding this is I thought the form will only be considered as complete when we get an ok response from third party system. 

>And if the user needed to change some information, they could click "Update info"?

Yeah.

**Comment by fourthletter** [comment]
## User story
1) A vendor submitting this form may have multiple contracts or already has an associated ID within Intacct. 2) Vendor is completing this vendor set-up form to override information in Intacct, e.g., banking information, address, or point of contact. 

## Description
[ ] Add a question to this set-up form to inquire whether this is a first-time contractor: "Did you previously receive funding from OTF?" "If yes, what was your contract number?"
[ ] Finance matches and verifies whether this is an existing vendor from a drop down list that is only available for Finance to view. 
[ ] Finance screens vendor to ensure vendor is a NEW vendor and that this request is from an officially approved project/point of contact/entity.

**Comment by sks444** [comment]
@frjo, let me know where in the form we would add these fields?

--

Some Notes:

We automatically use the same vendor for new projects if they already had a project.

One vendor can have multiple projects and same vendor info can be used on all projects.

**Comment by frjo** [comment]
@sks444 We need to clarify this with OTF, @ei8fdb will do that.

**Comment by sks444** [comment]
@frjo, is it possible to create a separate issue for the remaining requirements?

We can finalise and merge this pr in order to work on next tasks related to payment.

We also have the [invoicing pr](https://github.com/HyphaApp/hypha/pull/2450) rebased on top of vendor, ready for testing.

**Comment by fourthletter** [comment]
@sks444 Thank you for this update. I could test the **invoicing pr** this week.
